NOnion/Network/TorGuard.fs

```diff
@@ -96,50 +180,22 @@ type TorGuard private (client: TcpClient, sslStream: SslStream) =
     static member NewClientAsync ipEndpoint =
         TorGuard.NewClient ipEndpoint |> Async.StartAsTask
 
-    member __.Send (circuidId: uint16) (cellToSend: ICell) =
-        let safeSend() =
-            async {
-                use memStream = new MemoryStream(Constants.FixedPayloadLength)
-                use writer = new BinaryWriter(memStream)
-                cellToSend.Serialize writer
-
-                // Write circuitId and command for the cell
-                // (We assume every cell that is being sent here uses 0 as circuitId
-                // because we haven't completed the handshake yet to have a circuit
-                // up.)
-                try
-                    do!
-                        circuidId
-                        |> IntegerSerialization.FromUInt16ToBigEndianByteArray
-                        |> StreamUtil.Write sslStream
-
-                    do!
-                        Array.singleton cellToSend.Command
-                        |> StreamUtil.Write sslStream
-
-                    if Command.IsVariableLength cellToSend.Command then
-                        do!
-                            memStream.Length
-                            |> uint16
-                            |> IntegerSerialization.FromUInt16ToBigEndianByteArray
-                            |> StreamUtil.Write sslStream
-                    else
-                        Array.zeroCreate<byte>(
-                            Constants.FixedPayloadLength
-                            - int memStream.Position
-                        )
-                        |> writer.Write
-
-                    do! memStream.ToArray() |> StreamUtil.Write sslStream
-                with
-                | exn ->
-                    match FSharpUtil.FindException<SocketException> exn with
-                    | Some socketExn ->
-                        return raise <| NOnionSocketException socketExn
-                    | None -> return raise exn
-            }
+    member __.Send (circuitId: uint16) (cellToSend: ICell) =
+        async {
+            let! sendResult =
+                sendMailBox.PostAndAsyncReply(fun _replyChannel ->
```

Comment:
@aarani this arg is used, so it shouldnt have an underscore

Reply:
Fixed in c5c9b5e18c3ff4052145c3a160806b6bbae1a2b2.